openpype/hosts/nuke/plugins/load/load_clip.py

```diff
@@ -222,18 +222,21 @@ def _representation_with_hash_in_frame(self, representation):
         """
         representation = deepcopy(representation)
         context = representation["context"]
-        template = representation["data"]["template"]
+
+        # Get the frame from the context and hash it
+        frame = context["frame"]
```

Comment:
The `"frame" in context` was there because if `{originalBasename}` is use `"frame"` key won't be on context. So this would crash with `KeyError`. So it should be rather:
```
        representation = deepcopy(representation)
        context = representation["context"]
        template = representation["data"]["template"]
        if (
            "{originalBasename}" in template
            and "frame" in context
        ):
            frame = context["frame"]
            hashed_frame = "#" * len(str(frame))
            origin_basename = context["originalBasename"]
            context["originalBasename"] = origin_basename.replace(
                frame, hashed_frame
            )

            representation["context"]["frame"] = hashed_frame
```

Reply:
```
        if (
            "{originalBasename}" in template
            and "frame" in context
        ):
```
With what you're saying then this would never be true, no? Unless the template had `{frame}` as well?

Reply:
> The "frame" in context was there because if {originalBasename} is used, the "frame" key won't be on context.

@iLLiCiTiT  I haven't seen this usecase. Could you give me an example when this can happen? In Editorial it was always adding `frame` key to context data even `originalBasename` was set in publishing template. 

> With what you're saying then this would never be true, no? Unless the template had {frame} as well?
@BigRoy that is correct, this will never be true.

Reply:
Actually @iLLiCiTiT has a point. Here is the code which is removing frame from the template, right? https://github.com/ynput/OpenPype/blob/767d6af84d5f9cc975093a1ac6093146ec685357/openpype/plugins/publish/integrate.py#L660

I presume it is only removed from formating data in integrating but not from context data, or am I wrong?

Reply:
Anyway I hope it is clear that the pugins function `_representation_with_hash_in_frame` is only run in case the representation is image sequence. And to identify the representation as sequence there has to be more files under a representation. 

From the integration perspective it seems to me that we are adding representation with frame in context data allways in case if it is detected as image sequence.  So I do not see a problem with the code in the state as it is now.

Reply:
Ah, quickly looking at the integrator code then `originalBaseName` publishes could indeed still have `frame` - interesting. And they could even have `udim` too? Do they? Where do those then get collected I wonder?

Reply:
A udim sequence will get a value for `udim`, is still a sequence and would not have `frame`. I'd personally say the compositor should still be able to load it so that you could procedurally manipulate texture sets if needed.

Reply:
After personal discussion with @iLLiCiTiT we came to conclude that the case where `frame` would not in context.Data within clip loading might not happen and this is therefore mergable.